t/signer_expiration_set_time.t

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/perl -I../lib
+
+use strict;
+use warnings;
+use Test::Simple tests => 6;
+use Test::More;
+
+use Mail::DKIM::Signer;
+use Mail::DKIM::Verifier;
+
+my $homedir = ( -d "t" ) ? "t" : ".";
+
+my $tdir    = -f "t/test.key" ? "t" : ".";
+my $keyfile = "$tdir/test.key";
+
+sub generate_signed_email {
+  my ($timestamp,$expiration) = @_;
+
+  my $dkim    = Mail::DKIM::Signer->new(
+    Algorithm  => "rsa-sha256",
+    Method     => "relaxed",
+    Domain     => "example.org",
+    Selector   => "expirationtest",
+    KeyFile    => $keyfile,
+    Timestamp  => $timestamp,
+    Expiration => $expiration,
+  );
+  ok( $dkim, "new() works" );
+
+  my $sample_email = <<END_OF_SAMPLE;
+From: jason <jason\@example.org>
+Subject: hi there
+Comment: what is a comment
+
+this is a sample message
+END_OF_SAMPLE
+  $sample_email =~ s/\n/\015\012/gs;
+
+  $dkim->PRINT($sample_email);
+  $dkim->CLOSE;
+
+  my $signature = $dkim->signature;
+  ok( $signature, "signature() works" );
+
+  print "# signature=" . $signature->as_string . "\n";
+  ok( $signature->as_string =~ / t=$timestamp; /, "got expected signature timestamp value" );
+  ok( $signature->as_string =~ / x=$expiration; /, "got expected signature expiration value" );
+
+  my $signed_email = $signature->as_string . "\r\n" . $sample_email;
+  return $signed_email;
+}
+
+my $timestamp = time;
+my $expiration = $timestamp + 3600;
+my $signed_email = generate_signed_email($timestamp,$expiration);
+my $verifier = Mail::DKIM::Verifier->new();
+$verifier->{verify_time} = $timestamp + 86400;
+$verifier->PRINT($signed_email);
+$verifier->CLOSE;
+isnt( $verifier->result, 'pass', 'Expired Signature does not pass');
+
+$verifier = Mail::DKIM::Verifier->new();
+$verifier->PRINT($signed_email);
+$verifier->CLOSE;
+is( $verifier->result, 'pass', 'Non-expired Signature does pass');
+
+# override the DNS implementation, so that these tests do not
+# rely on DNS servers I have no control over
+my $CACHE;
+
+sub Mail::DKIM::DNS::fake_query {
```

Comment:
ditto

Reply:
This should/could all be replaced with passing Net::DNS::Resolver::Mock as the resolver, which would be much tidier and remove fake DNS code from the tests altogether, just pass in a zonefile and we're done.
We should check to make sure there are no Perl version issues, but I'm going to argue that this, and the previous tests cleanup comment should be their own task and MR.

Reply:
I'll take that as a reasonable answer.  In an ideal world that'd be done first but we don't live in one.